### PR TITLE
Update subjet handling

### PR DESCRIPTION
**Pull request by gkasieczka**
Updated JEC and uncertainty for subjets. 

Still need to test with 8_0_21.

### CODE REVIEW

**Comment by arizzi** [comment]
let me know once tested so that I can merge

**Comment by jpata** [comment]
@gkasieczka could you cherry-pick https://github.com/jpata/cmssw/commit/59f0257d4dc7cdf29eadb2f05ede0f88ff71f1df?

This solves the accidental reversion of the JetReCalibrator that we discussed with @arizzi privately. I'll make a separate PR against cbernet/heppy_8_0_19, but since vhbb is already 8_0_21, it's easier to make just one jet-related PR here, I think.

**Comment by arizzi** [comment]
can I merge then? or should I wait the cherry-pick?

**Comment by gkasieczka** [comment]
The cherry pick is in.

The output looks good, but I'm getting lots of errors from JetCorrectionUncertainty/SimpleJetCorrectionUncertainty at the moment.

**Comment by arizzi** [comment]
you mean this?

%MSG-e SimpleJetCorrectionUncertainty: 
 bin variables out of range
%MSG


I see that too... @jpata can you have a look? I see them also updating the GT

**Comment by jpata** [comment]
I started to have a look a little while ago

**Comment by gkasieczka** [comment]
Yes, that's what I mean. Though it does not seem to be caused by @jpata's extra lines.

File: PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py [hunk 1]
```diff
@@ -137,8 +137,8 @@ def getCorrection(self,jet,rho,delta=0,corrector=None,uncertainty="Total", recal
         JetUncertainty.setJetEta(jet.eta())
         JetUncertainty.setJetPt(corr * jet.pt() * jet.rawFactor())
 
-        try:
-            jet.jetEnergyCorrUncertainty = self.JetUncertainty.getUncertainty(True)
```
Comment (arizzi) [inline PhysicsTools/Heppy/python/physicsutils/JetReCalibrator.py:140]: right that was my quick hack this morning to get thing running, should not be needed with jpata fixes

**Comment by gkasieczka** [comment]
Latest commit (cc08685) fixes the issue

**Comment by jpata** [comment]
output seems OK.

File: VHbbAnalysis/Heppy/test/vhbb.py [hunk 2]
```diff
@@ -349,18 +349,36 @@
...
+                jet_type.variables += [
+                    NTupleVariable(
+                        "corr_{0}".format(name),
+                        lambda x, name = name: getattr(x, 'corr{0}'.format(name), -99),
+                        float,
+                        mcOnly=True,
+                        help=""
+                    )
+                ]
+# HTT Subjets
```
Comment (arizzi) [inline VHbbAnalysis/Heppy/test/vhbb.py:366]: does it make sense to have subjet stuff in vhbb.py rather than vhbb_combined.py?

**Comment by jpata** [comment]
The heppy-side version of the fix incorporated here is https://github.com/cbernet/cmssw/pull/82

File: VHbbAnalysis/Heppy/test/vhbb.py (hunk 2 above)
Comment (gkasieczka) [reply VHbbAnalysis/Heppy/test/vhbb.py:366]: I intially had it in vhbb.py as I didn't want to duplicate the factorizedJetCorrections list. However as we "from vhbb import *" anyways, we can also move.

**Pull request merged**